module/VuFind/src/VuFind/Captcha/ImageFactory.php

```diff
@@ -0,0 +1,96 @@
+<?php
+/**
+ * Factory for Image CAPTCHA module.
+ *
+ * PHP version 7
+ *
+ * Copyright (C) Villanova University 2020.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  CAPTCHA
+ * @author   Mario Trojan <mario.trojan@uni-tuebingen.de>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+namespace VuFind\Captcha;
+
+use Interop\Container\ContainerInterface;
+use Laminas\ServiceManager\Factory\FactoryInterface;
+
+/**
+ * Image CAPTCHA factory.
+ *
+ * @category VuFind
+ * @package  CAPTCHA
+ * @author   Mario Trojan <mario.trojan@uni-tuebingen.de>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+class ImageFactory implements FactoryInterface
+{
+    /**
+     * Create an object
+     *
+     * @param ContainerInterface $container     Service manager
+     * @param string             $requestedName Service being created
+     * @param null|array         $options       Extra options (optional)
+     *
+     * @return object
+     *
+     * @throws ServiceNotFoundException if unable to resolve the service.
+     * @throws ServiceNotCreatedException if an exception is raised when
+     * creating a service.
+     * @throws ContainerException if any other error occurs
+     */
+    public function __invoke(ContainerInterface $container, $requestedName,
+        array $options = null
+    ) {
+        if (!empty($options)) {
+            throw new \Exception('Unexpected options passed to factory.');
+        }
+
+        $imageOptions = [
+            'font' => getenv('VUFIND_HOME') . '/vendor/endroid/qr-code/assets/fonts/open_sans.ttf',
```

Comment:
What do you think about adding a composer package instead, e.g. webfontkit/open-sans?

Reply:
Sure, I could live with that!

Reply:
Done - if you still think we need a test class, can you please tell me where it should be located?

Reply:
You could make an ImageFactoryTest that extends [MockContainerTest](https://github.com/vufind-org/vufind/blob/master/module/VuFind/src/VuFindTest/Unit/MockContainerTest.php) and sets up a configuration like in [DoiLookupTest](https://github.com/vufind-org/vufind/blob/6a35282ed1730322b3ec1f36c0127bd2831b8a7a/module/VuFind/tests/unit-tests/src/VuFindTest/AjaxHandler/DoiLookupTest.php#L55). You could create a small, simple class to pass in to the factory as `$requestedName` which just returns the object passed in to the constructor, and then you can make some assertions about the options passed to the constructor. This will confirm that the factory parses/converts the configuration correctly... and you can add a `file_exists` check on the font setting for good measure.

If you don't have time to deal with this right now, feel free to create a TODO referencing this comment and I might have time to help with it later.

Reply:
This looks a bit confusing to me, so i would be grateful if you could tent to this, i already created a TODO.

Reply:
Sure, I'll see if I can do that right now!

Reply:
...done!